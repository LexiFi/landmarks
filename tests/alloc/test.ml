module L = Landmark

let name = "zero_alloc"
let zero_alloc = L.register name

let () =
  L.start_profiling
    ~profiling_options:{L.default_options with allocated_bytes = true} ();
  L.enter zero_alloc;
  L.exit zero_alloc

let check_allocated_bytes () =
  let open Landmark.Graph in

  let graph = L.export () in
  let node =
    root graph
    |> children graph
    |> List.filter (fun n -> n.name = name)
    |> List.hd
  in
  Printf.printf "Allocations:\n%d\n%d\n"
    (node.allocated_bytes)
    (node.allocated_bytes_major)

let () =
  check_allocated_bytes ()
