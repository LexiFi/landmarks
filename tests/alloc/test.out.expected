Allocations:
0
0
